SpatialGDK/Source/SpatialGDK/Private/Utils/LockingStatics.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/LockingStatics.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "LoadBalancing/AbstractLockingPolicy.h"
+#include "SpatialConstants.h"
+#include "Utils/SpatialStatics.h"
+
+#include "Containers/UnrealString.h"
+
+DEFINE_LOG_CATEGORY(LogLocking);
+
+namespace
+{
+bool CanProcessActor(const AActor* Actor)
+{
+	if (Actor == nullptr)
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on nullptr Actor is invalid."));
+		return false;
+	}
+
+	const UNetDriver* NetDriver = Actor->GetWorld()->GetNetDriver();
+	if (!NetDriver->IsServer())
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on a client is invalid. Actor: %s"), *GetNameSafe(Actor));
+		return false;
+	}
+
+	if (!Actor->HasAuthority())
+	{
+		UE_LOG(LogLocking, Error, TEXT("Calling locking API functions on a non-auth Actor is invalid. Actor: %s."),
+			*GetNameSafe(Actor));
+		return false;
+	}
+
+	return true;
+}
+} // anonymous namespace
+
+FLockingToken ULockingStatics::AcquireLock(AActor* Actor)
+{
+	if (!CanProcessActor(Actor) || !USpatialStatics::IsSpatialMultiWorkerEnabled(Actor->GetWorld()))
+	{
+		return FLockingToken{ SpatialConstants::INVALID_ACTOR_LOCK_TOKEN };
+	}
+
+	UAbstractLockingPolicy* LockingPolicy = Cast<USpatialNetDriver>(Actor->GetWorld()->GetNetDriver())->LockingPolicy;
+
+	const uint32 NewLockCount = LockingPolicy->GetActorLockCount(Actor) + 1;
+
+	const ActorLockToken LockToken = LockingPolicy->AcquireLock(Actor,
```

Comment:
Perhaps we just allow the user to pass the lock context string through?

Reply:
Yea, I kept changing what this looked like but I think ur right